dbms/src/Flash/Planner/Planner.cpp

```diff
@@ -112,21 +115,18 @@ void Planner::restorePipelineConcurrency(DAGPipeline & pipeline)
 void Planner::executeImpl(DAGPipeline & pipeline)
 {
     PhysicalPlan physical_plan{context, log->identifier()};
-    for (const auto & input_streams : input_streams_vec)
+    assert(query_block.children.size() == input_streams_vec.size());
+    for (size_t i = 0; i < input_streams_vec.size(); ++i)
     {
-        RUNTIME_ASSERT(!input_streams.empty(), log, "input streams cannot be empty");
-        physical_plan.buildSource(input_streams);
+        RUNTIME_ASSERT(!input_streams_vec[i].empty(), log, "input streams cannot be empty");
+        assert(query_block.children[i] && query_block.children[i]->root && query_block.children[i]->root->has_executor_id());
```

Comment:
It is not true for some list based executors(sent from TiSpark), do we plan to support that in Planner?

Reply:
for list based executors, input_streams_vec.size() == 0.
So no error will occur here for list based executors :)

Reply:
> do we plan to support that in Planner?
yes

Reply:
why list based executors has input_streams_vec.size() == 0 ? 
does it take another interpreter path?

Reply:
Because for list based executors , there will only be one query block.
And the source of query block will only be table scan.

Reply:
for TableScan, the input_streams_vec must be empty

Reply:
got